### PR TITLE
Progress on tests for AavegothciGameFacet and LibAavegotchi

### PR DESCRIPTION
**Pull request by solidigotchi**
Details

This PR implements progress on tests for both LibAavegotchi and AavegotchiGameFacet.

Checklist
[X] Created/updated tests.
 [] Created/updated documentation.
 [] Sensitive information not committed.
 [] Smoke tested with no evident errors.

### CODE REVIEW

**Comment by solidigotchi** [comment]
@mudgen Can you clarify which characters are to be supported for names? We may consider sanitizing them a bit more, as there are a number of characters in between the range in the description that may not be desired.

**Comment by solidigotchi** [comment]
@mudgen Also, the fix for xpUnitlNextLevel is solved with the conditional, but is there any consideration in using SafeMath?

**Comment by cinnabarhorse** [comment]
Hi there, thanks for the commit! @mudgen is offline right now but I'll let him know about these commits when he comes online. In the meantime feel free to DM me on TG (@coderdannn) or Discord!

**Comment by mudgen** [comment]
> @mudgen Also, the fix for xpUnitlNextLevel is solved with the conditional, but is there any consideration in using SafeMath?

I really appreciate the help!!!

We are using Solidity 0.8.1 which does checked math automatically, so no need for SafeMath.

**Comment by mudgen** [comment]
@solidigotchi 

> The checks against invalid characters used decimal instead of hex representations of characters. The assumption made was that that ascii characters 0x30-0x5a be used.

I don't think this matters.  In the end they are the same number whether using decimal (base 10)  or hex (base 16).

**Comment by mudgen** [comment]
Just the `20`  needed to be changed to `32`  the rest of the numbers in LibAavegotchi.validateAndLowerName are correct.  thanks I made this change.  This pull request was helpful, thanks.  In this case I'm not merging this pull request but I made the number change based on this pull request.

**Comment by solidigotchi** [comment]
> Just the `20` needed to be changed to `32` the rest of the numbers in LibAavegotchi.validateAndLowerName are correct. thanks I made this change. This pull request was helpful, thanks. In this case I'm not merging this pull request but I made the number change based on this pull request.

Cool; I'll pull downstream.

**Comment by mudgen** [comment]
> Cool; I'll pull downstream.

Okay,  I just pushed changes.

File: contracts/Aavegotchi/libraries/LibAavegotchi.sol [hunk 1]
```diff
@@ -229,6 +229,7 @@ library LibAavegotchi {
 
     function xpUntilNextLevel(uint256 _experience) internal pure returns (uint256 requiredXp_) {
         uint256 currentLevel = aavegotchiLevel(_experience);
+        if (currentLevel == 99) return 0;
```
Comment (solidigotchi) [inline contracts/Aavegotchi/libraries/LibAavegotchi.sol:232]: @mudgen Is this something you would like to fix upstream as well?

**Pull request closed without merging**